main.ts

```diff
@@ -282,19 +310,29 @@ class BeautitabPluginSettingTab extends PluginSettingTab {
 			.setDesc(
 				`Which plugin should be utilized for search when clicking the middle of the screen button?`
 			)
-			.addDropdown((component) => {
-				SearchProviders.forEach((provider) => {
-					component.addOption(provider.value, provider.display);
-				});
-
-				component.setValue(this.plugin.settings.inlineSearchProvider);
+			.setClass("search-provider")
 
-				component.onChange((value: SearchProvider) => {
-					this.plugin.settings.inlineSearchProvider = value;
-					this.plugin.settingsObservable.setValue(
-						this.plugin.settings
-					);
-					this.plugin.saveSettings();
+			.addText((component) => {
+				component
+					.setValue(this.plugin.settings.inlineSearchProvider.display ?? DEFAULT_SEARCH_PROVIDER.display)
```

Comment:
Shouldn't need the conditional because `this.plugin.settings.inlineSearchProvider.display` will always have a default value.